components/places/src/ffi.rs

```diff
@@ -398,7 +399,17 @@ impl PlacesConnection {
     // sync metadata to only sync after most recent visit to prevent
     // further syncing of older data
     fn delete_everything_history(&self) -> Result<()> {
-        self.with_conn(history::delete_everything)
+        let conn = self.db.lock();
+        let result = history::delete_everything(&conn);
+        if let Err(e) = &result {
+            if matches!(
```

Comment:
Maybe add a comment here about why that specific error gets that specific treatment - a link to the issue maybe?